InternTrack.Core.Api/Controllers/InternController.cs

```diff
@@ -100,5 +100,34 @@ public async ValueTask<ActionResult<Intern>> GetInternById(Guid internId)
                 return InternalServerError(internServiceException);
             }
         }
+
+        [HttpPut]
+        public async ValueTask<ActionResult<Intern>> PutInternAsync(Intern intern)
+        {
+            try
+            {
+                Intern modifiedIntern = 
+                    await this.internService.ModifyInternAsync(intern);
+
+                return Ok(modifiedIntern);
+            }
+            catch (InternValidationException internValidationException)
+                when (internValidationException.InnerException is NotFoundInternException)
+            {
+                return NotFound(internValidationException.InnerException);
+            }
+            catch (InternValidationException internValidationException)
+            {
+                return BadRequest(internValidationException.InnerException);
+            }
+            catch (InternDependencyException internDependencyException)
+            {
+                return Problem(internDependencyException.Message);
+            }
+            catch (InternServiceException internServiceException)
+            {
+                return Problem(internServiceException.Message);
```

Comment:
Instead of `Problem`, I would use InternalServerError in this case. I am ok with preview catch block, unless Ustadh @hassanhabib thinks otherwise.